fm.c

```diff
@@ -113,14 +111,10 @@ void FM_HandleIO(fm_t *chip)
 
 int FM_GetBus(fm_t *chip)
 {
-    int data = 0;
-    int io_dir = chip->input.cs && chip->input.rd && !chip->input.ic;
-    int data_enable = !io_dir && !chip->input.ic;
-    if (data_enable)
-        data = chip->data_latch;
-    if (chip->io_ic_latch[1])
-        data = 0;
-    return data;
+    if (!(chip->input.cs && chip->input.rd && !chip->input.ic) && !chip->input.ic)

```

Comment:
should be `chip->io_ic_latch[1]`?

Reply:
i suppose `!chip->io_ic_latch[1]` to verify it, i assumed i would need to check, my fault at translating the logic.